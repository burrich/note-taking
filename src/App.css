/* TODO : sass + flex mixin */

.App {
  display: flex;
  flex-direction: column;
  min-height: 100%;

  width: 90vw;
  margin: 0 auto;
}

.RichEditor-header h1 {
  text-align: center;
}

.RichEditor-root {
  flex-grow: 1;

  display: flex;
  flex-direction: column;
  min-height: 100%;
}

.RichEditor-styleButton {
  margin-right: 10px;
  cursor: pointer;
}

.RichEditor-editor {
  flex-grow: 1;

  display: flex;
  flex-direction: column;
  min-height: 100%;

  padding: 20px;
  margin-bottom: 40px;
  border-radius: 0 5px 5px 5px;
  background: white;
  cursor: text;
  /*border: 1px solid grey;*/
}

.DraftEditor-root {
  flex-grow: 1;

  display: flex;
  flex-direction: column;
  min-height: 100%;
}

.DraftEditor-editorContainer {
  flex-grow: 1;

  display: flex;
  flex-direction: column;
  min-height: 100%;
}

.public-DraftEditor-content {
  flex-grow: 1;
}

/* Code block */
/* TODO: change selectors */
.DraftEditor-root pre {
  /*border: 1px solid #ccc;*/
  background: #f2f2f2;
  border-radius: .2em;
  margin: 0;
  padding: .5em;
}

.DraftEditor-root pre > pre {
  background: none; /* ? */
  border: none;
  padding: 0;
}

