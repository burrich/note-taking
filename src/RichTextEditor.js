import React, {Component} from 'react';
import {EditorState, RichUtils} from 'draft-js';
import Editor from 'draft-js-plugins-editor';

import '../node_modules/draft-js/dist/Draft.css'; // TODO: move file outside node_modules
import './RichTextEditor.css';

/**
 * RichTextEditor class implementating draft.js RTE editor
 *
 * TODO: methods comments
 */
class RichTextEditor extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editorState: EditorState.createEmpty()
    };

    // this methods binding
    this.onChange = this.onChange.bind(this);
    this.focus    = this.focus.bind(this);
  }

  componentDidMount() {
    this.focus();
  }

  onChange(editorState) {
    this.setState({editorState});
  }

  focus() {
    this.domEditor.focus();
  }

  handleKeyCommand(command, editorState) {
    const newState = RichUtils.handleKeyCommand(editorState, command);
    if (newState) {
      this.onChange(newState);
      return 'handled';
    }
    return 'not-handled';
  }
  
  toggleStyleBlock(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'BOLD'));
  }

  toggleStyleItalic(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'ITALIC'));
  }

  toggleStyleUnderline(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'UNDERLINE'));
  }

  toggleStyleStrikethrough(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'STRIKETHROUGH'));
  }

  toggleStyleBlockquote(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'blockquote'));
  }

  toggleStyleCode(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'code-block'));
  }

  toggleStyleUl(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'unordered-list-item'));
  }

  toggleStyleOl(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'ordered-list-item'));
  }

  onTab(e) {
    const maxDepth = 4;
    this.onChange(RichUtils.onTab(e, this.state.editorState, maxDepth));
  }

  render() {
    // TODO: move h1 outside this component 
    return (
      <div className="RichEditor-root">
        <div className="RichEditor-controls">
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleBlock.bind(this)}>
            Bold
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleItalic.bind(this)}>
            Italic
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleUnderline.bind(this)}>
            Underline
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleStrikethrough.bind(this)}>
            Strikethrough
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleBlockquote.bind(this)}>
            Blockquote
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleCode.bind(this)}>
            Code
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleUl.bind(this)}>
            Ul
          </span>
          <span className="RichEditor-styleButton"
                onMouseDown={this.toggleStyleOl.bind(this)}>
            Ol
          </span>
        </div>

        <div className="RichEditor-editor" onClick={this.focus}>
          <Editor editorState={this.state.editorState}
                  onChange={this.onChange}
                  handleKeyCommand={this.handleKeyCommand.bind(this)}
                  onTab={this.onTab.bind(this)}
                  blockStyleFn={getBlockStyle}
                  spellCheck={true}
                  ref={el => this.domEditor = el} 
                  plugins={[]} />
        </div>
      </div>
    );
  }
}

function getBlockStyle(contentBlock) {
  const type = contentBlock.getType();
  if (type === 'blockquote') {
    return 'RichEditor-blockquote';
  }
}

export default RichTextEditor;
