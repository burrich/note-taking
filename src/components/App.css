.App {
  display: flex;
  flex-direction: column;
  min-height: 100%;

  width: 90vw;
  margin: 0 auto;
}

.App h1 {
  text-align: center;
}
