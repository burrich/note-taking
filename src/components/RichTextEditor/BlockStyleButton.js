import React from 'react';
import StyleButton from './StyleButton';

/**
 * BlockStyleButton component.
 */
function BlockStyleButton(props) {
  const style = styles[props.name];
  const active = props.editorStyle === style.code;
  
  return (
    <StyleButton 
      style={style}
      active={active}
      onToggle={props.onToggle}
      disabled={props.disabled} />
  );
}

const styles = {
  quotes: {
    code: 'blockquote',
    name: 'quote right'
  },
  code: {
    code: 'code-block',
    name: 'code'
  },
  ul: {
    code: 'unordered-list-item',
    name: 'unordered list'
  },
  ol: {
    code: 'ordered-list-item',
    name: 'ordered list'
  },
}

export default BlockStyleButton;
