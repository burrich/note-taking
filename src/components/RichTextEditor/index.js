import React, { Component } from 'react';
import { Editor, EditorState, ContentState, RichUtils, convertToRaw, convertFromRaw } from 'draft-js';
// import Editor from 'draft-js-plugins-editor';
import _ from 'lodash';

import '../../../node_modules/draft-js/dist/Draft.css'; // TODO: move file outside node_modules
import './styles/default.css';

import Controls from './Controls';

const LOG_TAG = '[RichTextEditor]';

/**
 * RichTextEditor component implementating draft.js RTE editor.
 * TODO: reafacto contructor, willReceive and createContent
 */
class RichTextEditor extends Component {
  constructor(props) {
    super(props);

    const note = this.props.note;
    const editorState = this.createContent(note);
    this.state = { editorState: editorState };

    this.domEditor = React.createRef();

    // this methods binding
    this.onChange          = this.onChange.bind(this);
    this.onTab             = this.onTab.bind(this);
    this.onFocus           = this.onFocus.bind(this);
    this.onBlur            = this.onBlur.bind(this);
    this.focus             = this.focus.bind(this);
    this.handleKeyCommand  = this.handleKeyCommand.bind(this);
    this.toggleInlineStyle = this.toggleInlineStyle.bind(this);
    this.toggleBlockStyle  = this.toggleBlockStyle.bind(this);
  }

  componentDidMount() {
    console.log(LOG_TAG, 'componentDidMount()');
    if (this.props.isFocus) {
      window.requestAnimationFrame(() => {
        console.log(LOG_TAG, 'focus');
        this.focus();
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    console.log(LOG_TAG, 'componentDidUpdate()');
    // Auto-save
    if (!this.props.disabled) {
      this.handleAutoSave(prevProps, prevState);
    }

    // Focus
    if (!prevProps.isFocus && this.props.isFocus) {
      // this.focus();
    }
  }

  onChange(editorState) {  
    this.setState({ editorState });
  
    // const rawContent = convertToRaw(editorState.getCurrentContent());
    // console.log(JSON.stringify(rawContent, null, 2));
  }

  onTab(e) {
    const maxDepth = 4;
    this.onChange(RichUtils.onTab(e, this.state.editorState, maxDepth));
  }

  onFocus(e) {
    if (!this.props.isFocus) {
      this.props.onFocus();
    }
  }

  onBlur(e) {
    if (this.props.isFocus) {
      this.props.onBlur();
    }
  }

  // TODO: better way to define the debounce method
  onUpdateContent = _.debounce((updatedNote) => {
    this.props.onSave(updatedNote);
  }, 1000);
  
  handleAutoSave(prevProps, prevState) {
    const previousRawContent = convertToRaw(prevState.editorState.getCurrentContent());
    const currentRawContent  = convertToRaw(this.state.editorState.getCurrentContent());
    // console.log('previousRawContent', previousRawContent);
    // console.log('currentRawContent', currentRawContent);

    const currentNote = this.props.note;
    const isContentUpdated = !_.isEqual(previousRawContent, currentRawContent);
    const hasNoteChanged = (currentNote !== prevProps.note) ? true : false;

    if (isContentUpdated && !hasNoteChanged) {
      const updatedNote = {
        ...currentNote,
        ...currentRawContent
      }
      console.log(LOG_TAG, 'onUpdateContent');
      this.onUpdateContent(updatedNote);
    }
  }

  handleKeyCommand(command, editorState) {
    const newState = RichUtils.handleKeyCommand(editorState, command);
    if (newState) {
      this.onChange(newState);
      return 'handled';
    }
    return 'not-handled';
  }

  createContent(note) {
    if (!note) {
      const defaultContent = ContentState.createFromText(
        'First you must add a note on the left side panel.'
      );
      return EditorState.createWithContent(defaultContent);
    }

    const contentState = convertFromRaw(note);
    const editorState = EditorState.createWithContent(contentState);
    return EditorState.moveSelectionToEnd(editorState); 
  }

  focus() {
    this.domEditor.current.focus();
  }
  
  getCurrentStyle() {
    const editorState = this.state.editorState;
    
    // get current editor inline style
    const currentInlineStyle = editorState.getCurrentInlineStyle();
    
    // get current editor block style
    const selection = editorState.getSelection(); 
    const currentBlockStyle = editorState
    .getCurrentContent()
    .getBlockForKey(selection.getStartKey())
    .getType();
    
    return {
      inline: currentInlineStyle,
      block: currentBlockStyle
    };
  }
  
  toggleInlineStyle(styleCode) {
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, styleCode));
  }

  toggleBlockStyle(styleCode) {
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, styleCode));
  }

  render() {
    console.log(LOG_TAG, 'render()');

    const editorState = this.state.editorState;
    const currentStyle = this.getCurrentStyle();
    const toggleStyle = {
      inline: this.toggleInlineStyle,
      block: this.toggleBlockStyle,
    };

    // TODO: state
    const readOnly = this.props.note ? false : true;
    let editorWrapperClass = 'RichEditor-editor';
    editorWrapperClass += !this.props.note ? ' RichEditor-editor-disabled' : '';

    return (
      <div className="RichEditor-root">
        <Controls currentStyle={currentStyle}
                  toggleStyle={toggleStyle}
                  editorDisabled={this.props.disabled} />

        <div className={editorWrapperClass} onClick={this.focus}>
          <Editor editorState={editorState}
                  onChange={this.onChange}
                  handleKeyCommand={this.handleKeyCommand}
                  onTab={this.onTab}
                  onFocus={this.onFocus}
                  onBlur={this.onBlur}
                  blockStyleFn={getBlockStyle}
                  spellCheck={true}
                  readOnly={readOnly}
                  ref={this.domEditor} />
                  {/*plugins={[]} />*/}
        </div>
      </div>
    );
  }
}

/**
 * Define CSS class to style blocks elements.
 */
function getBlockStyle(contentBlock) {
  const type = contentBlock.getType();
  if (type === 'blockquote') {
    return 'RichEditor-blockquote';
  }
}

export default RichTextEditor;
