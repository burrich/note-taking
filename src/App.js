import React, {Component} from 'react';
import {EditorState, RichUtils} from 'draft-js';
import Editor from 'draft-js-plugins-editor';

import '../node_modules/draft-js/dist/Draft.css'; // TODO: move file outside node_modules
import './App.css'; // TODO: rename file

/**
 * Draft RTE editor component class
 *
 * TODO: rename App
 * TODO: methods comments
 */
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editorState: EditorState.createEmpty()
    };

    this.onChange = (editorState) => this.setState({editorState});
    this.focus    = () => this.domEditor.focus();
  }

  componentDidMount() {
    this.domEditor.focus();
  }

  handleKeyCommand(command, editorState) {
    const newState = RichUtils.handleKeyCommand(editorState, command);
    if (newState) {
      this.onChange(newState);
      return 'handled';
    }
    return 'not-handled';
  }
  
  toggleStyleBlock(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'BOLD'));
  }

  toggleStyleItalic(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'ITALIC'));
  }

  toggleStyleUnderline(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'UNDERLINE'));
  }

  toggleStyleStrikethrough(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleInlineStyle(this.state.editorState, 'STRIKETHROUGH'));
  }

  toggleStyleBlockquote(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'blockquote'));
  }

  toggleStyleCode(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'code-block'));
  }

  toggleStyleUl(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'unordered-list-item'));
  }

  toggleStyleOl(e) {
    e.preventDefault();
    this.onChange(RichUtils.toggleBlockType(this.state.editorState, 'ordered-list-item'));
  }

  onTab(e) {
    const maxDepth = 4;
    this.onChange(RichUtils.onTab(e, this.state.editorState, maxDepth));
  }

  render() {
    // TODO: move h1 outside this component 
    return (
      <div className="App">
        <div className="RichEditor-header">
          <h1>Note taking</h1>
        </div>

        <div className="RichEditor-root">
          <div className="RichEditor-controls">
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleBlock.bind(this)}>
              Bold
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleItalic.bind(this)}>
              Italic
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleUnderline.bind(this)}>
              Underline
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleStrikethrough.bind(this)}>
              Strikethrough
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleBlockquote.bind(this)}>
              Blockquote
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleCode.bind(this)}>
              Code
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleUl.bind(this)}>
              Ul
            </span>
            <span className="RichEditor-styleButton"
                  onMouseDown={this.toggleStyleOl.bind(this)}>
              Ol
            </span>
          </div>

          <div className="RichEditor-editor" onClick={this.focus}>
            <Editor editorState={this.state.editorState}
                    onChange={this.onChange}
                    handleKeyCommand={this.handleKeyCommand.bind(this)}
                    onTab={this.onTab.bind(this)}
                    blockStyleFn={getBlockStyle}
                    spellCheck={true}
                    ref={el => this.domEditor = el} 
                    plugins={[]} />
          </div>
        </div>
      </div>
    );
  }
}

function getBlockStyle(contentBlock) {
  const type = contentBlock.getType();
  if (type === 'blockquote') {
    return 'RichEditor-blockquote';
  }
}

export default App;
