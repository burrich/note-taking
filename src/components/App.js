import React from 'react';
import RichTextEditor from './RichTextEditor';

import './App.css';

/**
 * App functionnal component
 *
 */
function App() {
  return (
    <div className="App">
      <h1>Note taking</h1>

      <RichTextEditor />
    </div>
  );
}

export default App;
