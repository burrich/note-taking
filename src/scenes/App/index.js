import React, { Component } from 'react';
import { Header } from 'semantic-ui-react';
import RichTextEditor from '../../components/RichTextEditor';
import NotesList from '../../components/NotesList';

import { getNotes, createNote, updateNote, deleteNote } from '../../services/api';

import './styles/default.css';

/**
 * App scene functionnal component.
 *
 */
class App extends Component {
  constructor(props) {
    super(props);

    this.state = {};

    getNotes((err, notes) => {
      if (err) return console.error(err);
      
      console.log(notes);

      this.setState({
        notes: notes,
        selectedNote: 0
      });
    });

    // this methods binding
    this.handleAddNote    = this.handleAddNote.bind(this);
    this.handleSelectNote = this.handleSelectNote.bind(this);
    this.handleEditNote   = this.handleEditNote.bind(this);
    this.handleRemoveNote = this.handleRemoveNote.bind(this);
    this.handleSaveNote   = this.handleSaveNote.bind(this);
  }

  handleAddNote(name) {
    // TODO: define newNote in RTE
    const newNote = {
      "name": name,
      "entityMap": {},
      "blocks": [
        {
          "key": "8i5rc",
          "text": "",
          "type": "unstyled",
          "depth": 0,
          "inlineStyleRanges": [],
          "entityRanges": [],
          "data": {}
        }
      ]
    }
    const newNoteToJson = JSON.stringify(newNote);

    createNote(newNoteToJson, (err, result) => {
      if (err) return console.error(err);

      console.log(result);

      // Update state with inserted id
      const updatedNotes = this.state.notes.slice();
      newNote._id = result.id;
      updatedNotes.push(newNote);

      this.setState({
        notes: updatedNotes,
        selectedNote: updatedNotes.length - 1
      });
    });

    // TODO: editor focus
  }

  handleSelectNote(index) {
    this.setState({ selectedNote: index });
  }

  handleEditNote(id, name) {
    const nameToJson = JSON.stringify({ name: name });

    updateNote(id, nameToJson, (err, result) => {
      if (err) return console.error(err);

      console.log(result);

      // Update state
      const updatedNotes = this.state.notes.slice();
      updatedNotes.find(elt => elt._id === id).name = name;

      this.setState({ notes: updatedNotes });
    });
  }

  handleRemoveNote(id) {
    deleteNote(id, (err, result) => {
      if (err) return console.error(err);

      console.log(result);

      // Update state
      // TODO: log immutability !!!
      const notes = this.state.notes;
      const updatedNotes = notes.filter(note => note._id !== id);
      
      let selectedNote = this.state.selectedNote;
      if (selectedNote !== 0) {
        selectedNote--;
      }

      this.setState({ 
        notes: updatedNotes,
        selectedNote: selectedNote
      });
    });
  }

  handleSaveNote(note) {
    const id = note._id;
    const rteAttrToJson = JSON.stringify({
      entityMap: note.entityMap,
      blocks: note.blocks
    });

    updateNote(id, rteAttrToJson, (err, result) => {
      if (err) return console.error(err);

      console.log(result);

      // Update state
      const updatedNotes = this.state.notes.slice();
      const index = updatedNotes.findIndex(elt => elt._id === id);

      updatedNotes[index] = note;
      this.setState({ notes: updatedNotes });
    });
  }

  render() {
    const notes = this.state.notes;
    if (!notes) {
      return (
        <div>Loadings notes...</div>
      );
    }

    const selectedNote = this.state.selectedNote;
    return (
      <div className="App">
        <div className="header">
          <Header
            as="h1"
            content="Note taking"
            textAlign="center" />
        </div>

        <div className="container-wrapper">
          <div className="container">
            <div className="container-left">
              {/* TODO: pass only notes names */}
              <NotesList 
                notes={notes}
                selectedNote={selectedNote}
                onAddNote={this.handleAddNote}
                onSelectNote={this.handleSelectNote}
                onEditNote={this.handleEditNote}
                onRemoveNote={this.handleRemoveNote} />
            </div>

            <div className="container-right">
              <RichTextEditor
                note={notes[selectedNote]}
                onSave={this.handleSaveNote} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
