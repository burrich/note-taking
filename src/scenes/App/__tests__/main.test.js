import React from 'react';
import { shallow } from 'enzyme'
import App from '../';
import RichTextEditor from '../../../components/RichTextEditor';

test('find one RichTextEditor', () => {
  const app = shallow(<App />);
  expect(app.find(RichTextEditor)).toHaveLength(1);
});
