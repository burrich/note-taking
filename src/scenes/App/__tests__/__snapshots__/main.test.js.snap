// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly with notes 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <Header
      as="h1"
      content="Note taking"
      textAlign="center"
    />
  </div>
  <div
    className="container-wrapper"
  >
    <div
      className="container"
    >
      <div
        className="container-left"
      >
        <NotesList
          notes={
            Array [
              Object {
                "blocks": Array [
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [
                      Object {
                        "length": 24,
                        "offset": 0,
                        "style": "BOLD",
                      },
                    ],
                    "key": "8i5rc",
                    "text": "Welcome to Note taking !",
                    "type": "unstyled",
                  },
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [],
                    "key": "a392a",
                    "text": "",
                    "type": "unstyled",
                  },
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [],
                    "key": "899no",
                    "text": "You are free to create your note, please enjoy ;)",
                    "type": "unstyled",
                  },
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [],
                    "key": "dd4je",
                    "text": "",
                    "type": "unstyled",
                  },
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [],
                    "key": "au5j1",
                    "text": "Burrich",
                    "type": "unstyled",
                  },
                ],
                "entityMap": Object {},
                "name": "Welcome",
              },
              Object {
                "blocks": Array [
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [],
                    "key": "5atf5",
                    "text": "foo",
                    "type": "unstyled",
                  },
                ],
                "entityMap": Object {},
                "name": "Note 1",
              },
              Object {
                "blocks": Array [
                  Object {
                    "data": Object {},
                    "depth": 0,
                    "entityRanges": Array [],
                    "inlineStyleRanges": Array [],
                    "key": "5atf5",
                    "text": "bar",
                    "type": "unstyled",
                  },
                ],
                "entityMap": Object {},
                "name": "Note 2",
              },
            ]
          }
          onAddNote={[Function]}
          onEditNote={[Function]}
          onRemoveNote={[Function]}
          onSelectNote={[Function]}
          selectedNote={0}
        />
      </div>
      <div
        className="container-right"
      >
        <RichTextEditor
          disabled={false}
          isFocus={true}
          note={
            Object {
              "blocks": Array [
                Object {
                  "data": Object {},
                  "depth": 0,
                  "entityRanges": Array [],
                  "inlineStyleRanges": Array [
                    Object {
                      "length": 24,
                      "offset": 0,
                      "style": "BOLD",
                    },
                  ],
                  "key": "8i5rc",
                  "text": "Welcome to Note taking !",
                  "type": "unstyled",
                },
                Object {
                  "data": Object {},
                  "depth": 0,
                  "entityRanges": Array [],
                  "inlineStyleRanges": Array [],
                  "key": "a392a",
                  "text": "",
                  "type": "unstyled",
                },
                Object {
                  "data": Object {},
                  "depth": 0,
                  "entityRanges": Array [],
                  "inlineStyleRanges": Array [],
                  "key": "899no",
                  "text": "You are free to create your note, please enjoy ;)",
                  "type": "unstyled",
                },
                Object {
                  "data": Object {},
                  "depth": 0,
                  "entityRanges": Array [],
                  "inlineStyleRanges": Array [],
                  "key": "dd4je",
                  "text": "",
                  "type": "unstyled",
                },
                Object {
                  "data": Object {},
                  "depth": 0,
                  "entityRanges": Array [],
                  "inlineStyleRanges": Array [],
                  "key": "au5j1",
                  "text": "Burrich",
                  "type": "unstyled",
                },
              ],
              "entityMap": Object {},
              "name": "Welcome",
            }
          }
          onBlur={[Function]}
          onFocus={[Function]}
          onSave={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
